src/main/java/pl/edu/agh/ki/io/db/UserStorage.java

```diff
@@ -17,13 +22,19 @@ public UserStorage(UserRepository userRepository, PasswordEncoder passwordEncode
         this.passwordEncoder = passwordEncoder;
     }
 
-    public User createOrUpdateUser(User user) {
+    public UserDetails createUser(User user) {
+        Optional<User> optionalUser = userRepository.findUserByEmail(user.getEmail());
+        if (optionalUser.isPresent()){
+            User storedUser = optionalUser.get();
```

Comment:
Chyba bym wywalil to przypisanie i od razu wywalil error z `optionalUser.get.()getAuthProvider()`

Reply:
dzięki, zapomniałem to zmienić